Add glossary (#3569)
Napari/napari: docs/glossary.md

| napari term        | Description                                                                                                                                                                                                                                                                                                                                 |
|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | channel            | A channel is a grayscale image. In fluorescence microscopy,  each channel corresponds to a range of wavelengths. An RGB image from digital camera consists of 3 channels - red, green, blue.                                                                                                                                                 |

Reviewer: ```suggestion
| channel            | A channel is a grayscale image. In fluorescence microscopy,  each channel corresponds to a range of wavelengths. An RGB image from digital camera consists of 3 channels - red, green, blue (RGB).                                                                                                                                                 |
```

> | colormap          | An array of colors used to map pixel data to the actual color values.                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| colormap          | An array of colors used to map pixel data to the actual color values. Also known as look-up table (LUT).                                                                                                                                                                                                                                                                      |
```

> | dock widget        | A widget that can be attached to the main napari viewer at a specified position.                                                                                                                                                                                                                                                            |

Reviewer: ```suggestion
| dock widget        | A window-like panel that can be attached to the main napari viewer at a specified position.                                                                                                                                                                                                                                                            |
```
We should not explain "widget" with "widget".

> | hook               | Hook is the place within napari where we “promise” to call functions created by external developers and installed by the user.                                                                                                                                                                                                              |

Reviewer: ```suggestion
| hook               | Hook is the place within napari where we “promise” to call functions created by external developers and installed by the user. Plugin developers can use hooks to extend functionality of napari.                                                                                                                                                                                                             |
```

> | hook specification | A function signature that defines the API that a plugin developer must adhere to when writing the function that we promise to call somewhere in the napari codebase.                                                                                                                                                                        |

Reviewer: ```suggestion
| hook specification | A function signature that defines the application programming interface (API) that a plugin developer must adhere to when writing the function that we promise to call somewhere in the napari codebase.                                                                                                                                                                        |
```

> | keybinding         | The key or key combination that causes some functions to occur.                                                                                                                                                                                                                                                                             |

Reviewer: ```suggestion
| keybinding         | The key or key combination that causes some functions to occur. Also known as keyboard short-cuts.                                                                                                                                                                                                                                                                            |
```

> | labels             | The labels layer is an image where pixel values are integers. Each unique integer represents a unique object, feature or classification.                                                                                                                                                                                                    |

Reviewer: ```suggestion
| labels             | The labels layer is an image where pixel values are integers. Each unique integer represents a unique object, feature or classification. See also the [napari labels layer tutorial](https://napari.org/tutorials/fundamentals/labels.html).                                                                                                                                                                                                   |
```

| layer              | Layers are the viewable objects that can be added to napari viewer. Currently napari supports the following layers: image, labels, points, shapes, surface, tracks, vectors.                                                                                                                                                                |

> | mousebinding       | The specific mouse click that causes some functions to occur.                                                                                                                                                                                                                                                                               |

Reviewer: In other programming environments such as C++ Qt, Windows Forms, etc, the terminology is "mouse-events", such as `onMousePress`, `onMouseMove`, `onMouseRelease`. Can we connect napari terminology to the terminology of those common platforms?

> | plugin             | A napari plugin is a specialized Python package which extends napari's functionality through additional file i/o support, image analysis, or other utilities. Plugin developers specify this functionality through a combination of Python entry points and a specialized manifest file. Plugins for napari can be found at napari-hub.org. |

Reviewer: We should mention `hooks` here, no?

> | shapes             | The shapes layer allows you to display a list of NxD arrays, where each array corresponds to one shape, specified by N points in D coordinates. Currently supported shapes: rectangle, ellipses, polygons, paths, and lines. Can be used as a visualization tool to highlight ROI, or be used for downstream operations.                    |

Reviewer: ```suggestion
| shapes             | The shapes layer allows you to display a list of NxD arrays, where each array corresponds to one shape, specified by N points in D coordinates. Currently supported shapes: rectangle, ellipses, polygons, paths, and lines. Can be used as a visualization tool to highlight ROI, or be used for downstream operations. See also the [napari shapes layer tutorial](https://napari.org/tutorials/fundamentals/shapes.html)                   |
```

> | surface            | A surface comprises a set of triangles (typically in three dimensions) that are connected by their common edges or corners to form the mesh.                                                                                                                                                                                                |

Reviewer: ```suggestion
| surface            | A surface comprises a set of triangles (typically in three dimensions) that are connected by their common edges or corners to form the mesh. [See also Wikipedia](https://en.wikipedia.org/wiki/Surface_triangulation) and the [napari surface layer tutorial](https://napari.org/tutorials/fundamentals/surface.html).                                                                                                                                                                                              |
```

> | widget             | A small, composable graphical user interface . A widget could be a simple one parameter/control (e.g. a slider to adjust contrast limits, a drop-down box to select a mode) or a collection of multiple controls with data output.                                                                                                          |

Reviewer 2: ```suggestion
| widget             | A small, composable graphical user interface component. A widget could be a simple one parameter/control (e.g. a slider to adjust contrast limits, a drop-down box to select a mode) or a collection of multiple controls with data output.                                                                                                          |
```

> | widget             | A small, composable graphical user interface component. A widget could be a simple one parameter/control (e.g. a slider to adjust contrast limits, a drop-down box to select a mode) or a collection of multiple controls with data output.                                                                                                          |

Reviewer: ```suggestion
| widget             | A small, composable graphical user interface component. A widget could be a simple one parameter/control (e.g. a button) or a collection of multiple controls with data output.                                                                                                          |
```

> | napari term        | description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | synonym/closely related term                                                |

Reviewer 3: For JupyterBook to recognize your document, it must have a title.

```suggestion
# Glossary

| napari term        | description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | synonym/closely related term                                                |
```

In addition, it needs to be added to the table of contents `_toc.yml` if you want to preview it locally, or if you want to see it in the historical docs pages hosted at napari.org/docs - I don't think that's critical here as the actual table of contents rendered in the final napari.github.io site is defined in the napari.github.io repo, not here. Once this document is ready we can add it there.

| hook               | Hook is the place within napari where we promise to call functions created by external developers and installed by the user. Plugin developers can use hooks to extend functionality of napari. See [this wiki page on hooking](https://en.wikipedia.org/wiki/Hooking) for additional details.                                                                                                                                                                                                                                                                                               |                                                                             |

> | hook specification | A function signature that defines the application programming interface (API) that a plugin developer must adhere to when writing the function that napari promises to call somewhere in the napari codebase. See also [napari hook specifications](https://napari.org/plugins/stable/hook_specifications.html).                                                                                                                                                                                                                                                                             |                                                                             |

Reviewer 3: To use the local sphinx links/references instead of a raw url, you can do
```suggestion
| hook specification | A function signature that defines the application programming interface (API) that a plugin developer must adhere to when writing the function that napari promises to call somewhere in the napari codebase. See also {ref}`napari hook specifications <hook-specifications-reference>`.                                                                                                                                                                                                                                                                             |                                                                             |
```

| canvas             | The canvas in the center of napari viewer contains the visual display of the data passed to napari, including images, points, shapes, and other supported data types.                                                                                                                                                                                                                                                                                                                                                                                                                        |                                                                             |

> | channel            | A channel is an image that each pixel only contains its intensity value (i.e. grayscale image). In fluorescence microscopy, each channel corresponds to a range of wavelengths. An RGB image from digital camera consists of 3 channels - red, green, blue (RGB). In napari, an image dimension that is not spatial or temporal can be considered as channels, and is typically represented as individual [image layer](https://napari.org/tutorials/fundamentals/image.html). See also [this wiki page](https://en.wikipedia.org/wiki/Channel_(digital_image)) for specific examples.       | raster band in geoscience                                                   |

Reviewer 4: ```suggestion
| channel            | A channel is an image where each pixel only contains its intensity value (i.e. grayscale image). In fluorescence microscopy, each channel corresponds to a range of wavelengths. An RGB image from digital camera consists of 3 channels - red, green, blue (RGB). In napari, an image dimension that is not spatial or temporal can be considered as channels, and is typically represented as individual [image layer](https://napari.org/tutorials/fundamentals/image.html). See also [this wiki page](https://en.wikipedia.org/wiki/Channel_(digital_image)) for specific examples.       | raster band in geoscience                                                   |
```

Reviewer 4: maybe "for which" is better grammar, not sure, but i don't think "that" is right

Reviewer 5: "where" sounds good to me

> | colormap           | A colormap represents the value of each pixel as a particular color. In fluorescence microscopy, the colormap can be selected to mirror the color of the collected wavelength. In other applications, colormaps can be used to provide the visual representation of the measured signal. There are many different types of colormaps available, such as [matplotlib colormap](https://matplotlib.org/stable/tutorials/colors/colormaps.html).                                                                                                                                                                                                                                                                                                                                                                    | look up table, LUT                                                          |

Reviewer 5: ```suggestion
| colormap           | A colormap represents the value of each pixel as a particular color. In fluorescence microscopy, the colormap can be selected to mirror the color of the collected wavelength. In other applications, colormaps can be used to provide the visual representation of the measured signal. There are many different types of colormaps available, see [choosing colormaps in matplotlib](https://matplotlib.org/stable/tutorials/colors/colormaps.html).                                                                                                                                                                                                                                                                                                                                                                    | look up table, LUT                                                          |
```

> | hook               | Hook is the place within napari where we promise to call functions created by external developers and installed by the user. Plugin developers can use hooks to extend functionality of napari. See [this wiki page on hooking](https://en.wikipedia.org/wiki/Hooking) for additional details.                                                                                                                                                                                                                                                                                               |                                                                             |

Reviewer 5: ```suggestion
| hook               | Hooks are a way to tell napari "If this event happens, then call this function". [Hooking up your own events](https://napari.org/docs/dev/guides/connecting_events.html?highlight=hook). Plugin developers can also use hooks to extend functionality of napari. See the [napari hook specification reference](https://napari.org/docs/dev/plugins/hook_specifications.html) in the [napari plugin documentation](https://napari.org/docs/dev/plugins/index.html).                                                                                                                                                                                                                                                                                               |                                                                             |
```

Reviewer 5: Reasons for suggestion:
1. Hooks are about *when* something happens, not *where*. I think it's much less confusing to describe it in those terms.
2. I think we should link to the napari docs pages on hooks & hookspecs, instead of Wikipedia. That's because for this concept, the reader will want to know how it applies specifically to napari (as opposed to the more general image analysis terms, where it does make sense to link to Wikipedia for a more general understanding, here it's unhelpful if the reader has only a general understanding but doesn't know how to apply that in napari).

> | hook specification | A function signature that defines the application programming interface (API) that a plugin developer must adhere to when writing the function that napari promises to call somewhere in the napari codebase. See also {ref}`napari hook specifications <hook-specifications-reference>`.                                                                                                                                                                                                                                                                             |                                                                             |

Reviewer 5: This reference seems broken. This is the page we want to point to: https://napari.org/docs/dev/plugins/hook_specifications.html

> | keybinding         | The specific key or key combination that when pressed causes some function to occur. For example, command/control + C triggers "copy" in many operating systems.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | keyboard short cut                                                          |

Reviewer 5: ```suggestion
| keybinding         | The specific key or key combination that when pressed causes some function to occur. For example, command/control + C triggers "copy" in many operating systems. You can make your own custom keybindings, see this page on [hooking up your own events](https://napari.org/docs/dev/guides/connecting_events.html).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | keyboard short cut                                                          |
```

> | mousebinding       | The specific mouse move, click, or drag event that causes some function to occur.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | mouse short cut                                                             |

Reviewer 5: ```suggestion
| mousebinding       | The specific mouse move, click, or drag event that causes some function to occur. You can make your own custom mousebindings, see this page on [hooking up your own events](https://napari.org/docs/dev/guides/connecting_events.html?highlight=hook).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | mouse short cut                                                             |
```

| layer              | Layers are the viewable objects that can be added to napari viewer. Currently napari supports the following layers: [image](https://napari.org/tutorials/fundamentals/image.html), [labels](https://napari.org/tutorials/fundamentals/labels.html), [points](https://napari.org/tutorials/fundamentals/points.html), [shapes](https://napari.org/tutorials/fundamentals/shapes.html), [surface](https://napari.org/tutorials/fundamentals/surface.html), [tracks](https://napari.org/tutorials/fundamentals/tracks.html), [vectors](https://napari.org/tutorials/fundamentals/vectors.html). |                                                                             |

> | label              | A label is a region of an integer image where the pixels are the same unique value. It is implemented in napari in the [labels layer](https://napari.org/tutorials/fundamentals/labels.html) where each label represents a unique object, feature or classification.                                                                                                                                                                                                                                                                                                                         |                                                                             |

Reviewer 5: :+1:

> | point              | A point is defined by the coordinate in space. [Points layer](https://napari.org/tutorials/fundamentals/points.html)  allows you to display an NxD array of N points in D coordinates.                                                                                                                                                                                                                                                                                                                                                                                                       |                                                                             |

Reviewer 5: ```suggestion
| point              | A point is defined by the coordinate in space. The [points layer](https://napari.org/tutorials/fundamentals/points.html)  allows you to display an NxD array of N points in D coordinates.                                                                                                                                                                                                                                                                                                                                                                                                       |                                                                             |
```

| surface            | A surface comprises a set of triangles (typically in three dimensions) that are connected by their common edges or corners to form the mesh. See [napari surface layer tutorial](https://napari.org/tutorials/fundamentals/surface.html) for current implementation and [this wiki page](https://en.wikipedia.org/wiki/Surface_triangulation) for additional information.                                                                                                                                                                                                                    |                                                                             |

> | track              | A track describes object trajectories along the time dimension. [Tracks layer](https://napari.org/tutorials/fundamentals/tracks.html) defines a track by the list containing object coordinates and the time point, with each track assigned a unique track ID.                                                                                                                                                                                                                                                                                                                              |                                                                             |

Reviewer 5: ```suggestion
| track              | A track describes object trajectories along the time dimension. The [tracks layer](https://napari.org/tutorials/fundamentals/tracks.html) defines a track by the list containing object coordinates and the time point, with each track assigned a unique track ID.                                                                                                                                                                                                                                                                                                                              |                                                                             |
```

> | vector             | A vector is a line defined by its start and end positions in space. [Vectors layer](https://napari.org/tutorials/fundamentals/vectors.html) renders lines onto the canvas, with input data format as Nx2xD numpy array representing N vectors in D dimensions.                                                                                                                                                                                                                                                                                                                               |                                                                             |

Reviewer 5: ```suggestion
| vector             | A vector is a line defined by its start and end positions in space. The [vectors layer](https://napari.org/tutorials/fundamentals/vectors.html) renders lines onto the canvas, with input data format as Nx2xD numpy array representing N vectors in D dimensions.                                                                                                                                                                                                                                                                                                                               |                                                                             |
```

| dock widget        | A widget that can be attached (docked) to the main napari viewer at a specified position.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |                                                                             |

> | plugin             | A napari plugin is a specialized Python package that extends napari's functionality. Examples of extended functionality include: additional file i/o support, image analysis, and other utilities. Plugin developers specify this functionality through a combination of Python entry points and a specialized manifest file. Plugins for napari can be found at [napari hub](https://www.napari-hub.org/).                                                                                                                                                                                 |                                                                             |

Reviewer 5: ```suggestion
| plugin             | A napari plugin is a specialized Python package that extends napari's functionality. Examples of extended functionality include: additional file i/o support, image analysis, and other utilities. Plugin developers specify this functionality through a combination of Python entry points and a specialized manifest file. Plugins for napari can be found at [napari hub](https://www.napari-hub.org/). For a guide on how to make your own custom plugin, see [creating a napari plugin[](https://napari.org/docs/dev/plugins/for_plugin_developers.html#plugins-for-plugin-developers).                                                                                                                                                                                 |                                                                             |
```
